src/main/java/edu/harvard/iq/dataverse/dataverse/banners/BannerDAO.java

```diff
@@ -38,6 +43,52 @@ public DataverseBanner getBanner(Long bannerId) {
         return em.find(DataverseBanner.class, bannerId);
     }
 
+    /**
+     * Fetches banners that are meant to be displayed for the dataverse
+     * (which also means dataverse banners that are higher in the *tree*)
+     *
+     * @param dataverseId
+     * @return ImageWithLinkDto
+     */
+    public List<ImageWithLinkDto> getBannersForDataverse(Long dataverseId) {
+        List<Object[]> banners = em.createNativeQuery("select r.image, r.imagelink from (select distinct dvtml.image, dvtml.imagelink, dvtm.totime  from\n" +
+                "  dataversebanner dvtm\n" +
+                "  join dataverselocalizedbanner dvtml on dvtml.dataversebanner_id = dvtm.id\n" +
+                "  where\n" +
+                "    dvtm.active = true and\n" +
+                "    dvtml.locale = ? and\n" +
+                "    ? between dvtm.fromtime and dvtm.totime and\n" +
+                "    dvtm.dataverse_id in (with recursive dv_roots as (\n" +
+                "    select\n" +
+                "        dv.id,\n" +
+                "        dv.owner_id,\n" +
+                "        d2.allowmessagesbanners\n" +
+                "    from dvobject dv\n" +
+                "      join dataverse d2 on dv.id = d2.id\n" +
+                "    where\n" +
+                "        dv.id = ?\n" +
+                "        union all\n" +
+                "        select\n" +
+                "               dv2.id,\n" +
+                "               dv2.owner_id,\n" +
+                "               d2.allowmessagesbanners\n" +
+                "        from dvobject dv2\n" +
+                "               join dataverse d2 on dv2.id = d2.id\n" +
+                "               join dv_roots on dv_roots.owner_id = dv2.id\n" +
```

Comment:
The aliases naming are really confusing `dv`, d2`, `dv2`. It is really hard to catch
up what is what.
Can we name it differently?
Maybe something along the lines

`dv_obj` - for outer dvobject (currently `dv`)

`dv` - for outer dataverse (currently `d2`)

`parent_dv_obj` - for inner dvobject (currently `dv2`)

`parent_dv` - for inner dataverse (currently also `d2`)

What do you think?

Reply:
Yeah that's true, fixed name's sounds fine.